### ISSUE TITLE
The Anchors position its broken in the extension part

### ISSUE THREAD

**Issue by Tarobish** [Tarobish/Jomhuria#43]
It works perfectly in live testing, this is something wrong with the Adobe. its broken in indesign
![87ff772a-df9c-11e5-84b5-dc8c3de75e95](https://cloud.githubusercontent.com/assets/8403973/13439846/257a87e0-dfa4-11e5-9695-677a1ba1d030.png)
![screen shot 2016-03-01 at 11 03 39 am]
![screen shot 2016-03-01 at 11 03 39 am](https://cloud.githubusercontent.com/assets/8403973/13439878/4747d38c-dfa4-11e5-8be5-5220c4def368.png)

**Comment by graphicore** [comment]
Looks also like missing anchors. The live testing link would still help a lot, even if it works there. So I can check it out myself.

**Comment by Tarobish** [comment]
:)

http://tarobish.github.io/Jomhuria/#live?eyJ2YWx1ZSI6Itir2Y7YpyDZvtmQ2ZHbjCDbjNmR2ZDbjCIsImJpZGkiOiJydGwiLCJsYW5nIjoiYXIifQ==

**Comment by graphicore** [comment]
![selection_086](https://cloud.githubusercontent.com/assets/393132/13448275/eeb752dc-e021-11e5-9df7-2457831f6cdc.png)

This is the result from shaping the live testing input—just to proof my claim that it helps ;-)

```
`uni06CC.fina|uni0650|uni0651|uni0640.1|uni06CC.init|space|uni06CC.fina|uni0650|uni0651|uni0640.1|uni067E.init|space|uni0627.fina|uni064E|uni0640.1|uni062B.init
```

The pieces `uni0640.1` are our Kashida, that we insert to stop glyph collisions. I believe we had the very same issue just recently in Mirza and Katibeh: https://github.com/Tarobish/Mirza/issues/92

I think the anchors in `uni0640.1` are missing for Adobe, which got confused by us inserting the Kashida (Harfbuzz handles this by using the original glyphs for the anchors, that's why it works in the browsers.)

Please add anchors to `uni0640.1`

**Comment by Tarobish** [comment]
You are always right :) just added. i'll check it later :)

**Comment by graphicore** [comment]
please check.

**Comment by Tarobish** [comment]
Here is the problem, characters are using same anchors be cause of the keshideh. it works somewhere it doesn't somewhere else 

![screen shot 2016-03-01 at 7 24 25 pm](https://cloud.githubusercontent.com/assets/8403973/13449830/e4d123d0-dfe3-11e5-848c-51eed5debf32.png)

**Comment by graphicore** [comment]
It may be that this works better in Mirza and Katibeh because of the decomposition we do there. In your example, the three dots are part of the first glyph, and not base + mark. So we destroy the mark-to-mark relation in Adobe by inserting the kashida. :-/ That's a bummer.  I'll have to think about this.

**Comment by Tarobish** [comment]
yeah, it works better in Mirza and katibeh. is there anyway to ignore marks from the Kesideh parts?
In that case, it works they way it has to be

**Comment by graphicore** [comment]
> is there anyway to ignore marks

I don't know what you mean. They are there, so they must go somewhere.

> In that case, it works they way it has to be 

I don't get it :-)

**Comment by Tarobish** [comment]
I know :) i mean generating marks from the right characters instead of keshideh
The problem is, the marks right now are generating from the keshideh part not from the characters thats why the position its wrong

**Comment by graphicore** [comment]
Yeah. That's what Adobe does different than Harfbuzz. When we insert that Kashideh, the shaping engine must decide where the marks belong. For us, Harfbuzz decides right and Adobe decides wrong.

**Comment by Tarobish** [comment]
Got it :)

**Comment by davelab6** [comment]
https://github.com/googlefonts/gf-docs/blob/master/ProjectChecklist.md#adobe-creative-suite

**Comment by graphicore** [comment]
@davelab6 Yeah I can report it. It's interesting though how it was possible to work around it in Mirza/Katibeh, but how it is hard here.

**Comment by davelab6** [comment]
https://github.com/twardoch/ttfdiet/ also full of 'no' for CC

**Comment by graphicore** [comment]
It's interesting that we don't have bigger problems with Adobe CC in Mirza/Katibeh, because we really do a lot of decomposition. But it's mostly done in init/medi/fina, maybe that's the silver bullet.
